pkg/container/container.go

```diff
@@ -296,6 +297,53 @@ func (c *Module) Run(ns string, data pkg.Container) (id pkg.ContainerID, err err
 	return pkg.ContainerID(container.ID()), nil
 }
 
+// Exec executes a command inside the container
+func (c *Module) Exec(ns string, containerID string, timeout time.Duration, args ...string) error {
+	client, err := containerd.New(c.containerd)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	ctx = namespaces.WithNamespace(ctx, ns)
+
+	container, err := client.LoadContainer(ctx, containerID)
+	if err != nil {
+		return errors.Wrap(err, "couldn't load container")
+	}
+	t, err := container.Task(ctx, nil)
+	if err != nil {
+		return errors.Wrap(err, "failed to create task")
+	}
+	var p specs.Process
+	p.Cwd = "/"
+	p.Args = args
+	taskID, err := uuid.NewUUID()
+	if err != nil {
+		return errors.Wrap(err, "failed to generate a uuid")
+	}
+	pr, err := t.Exec(ctx, taskID.String(), &p, cio.NullIO)
+	if err != nil {
+		return errors.Wrap(err, "failed to exec new porcess")
+	}
+	if err := pr.Start(ctx); err != nil {
+		return errors.Wrap(err, "failed to start process")
+	}
+	ch, err := pr.Wait(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to wait for process")
+	}
+	<-ch
```

Comment:
To avoid getting stuck forever, u also need to select between this and the ctx.Done() channel to make sure it's not stuck forever.

```suggestion
	select {
	  case <-ch:
	  case <-ctx.Done():
	    // force cancelation of the task
```